src/components/Projects/Members/Member/Member.test.jsx

```diff
@@ -77,38 +86,42 @@ describe('Member Component', () => {
 
   it('calls assignProject function with "unAssign" when the unassign button is clicked', async () => {
     const assignProject = jest.fn();
-    const { getByRole } = renderMemberRow({ sampleMember, assignProject });
-    
+    const hasPermission = jest.fn((a) => true)
+    sampleMember.assignProject = assignProject;
+    sampleMember.hasPermission = hasPermission;
+    const { getByRole } = renderMemberRow(sampleMember);
+
     // Simulate a button click on the actual button instead of the icon
     // const unassignButton = container.querySelector('.btn.btn-outline-danger.btn-sm');
     const unassignButton = getByRole('button');
     // const unassignButton = screen.queryByRole('button', { name: /unassign/i });
 
     expect(unassignButton).toBeInTheDocument();
     if (unassignButton) {
-        fireEvent.click(unassignButton);
+      fireEvent.click(unassignButton);
     }
-  
+
     // assignProject.mock.calls.forEach(call => {
     //   console.log(call);
     // });
-  
+
+    waitFor(() => {
+      expect(assignProject).toBeCalled();
+    });
+    // Verify that the assignProject function is called with the expected arguments
     waitFor(() => {
-        expect(assignProject).toBeCalled();
-      });
-      // Verify that the assignProject function is called with the expected arguments
-      waitFor(() => {
-        expect(assignProject).toHaveBeenCalledWith(
-          'project123',
-          'member123',
-          'unAssign',
-          'Jane',
-          'Doe'
-        );
-      });       
+      expect(assignProject).toHaveBeenCalledWith(
+        'project123',
+        'member123',
+        'unAssign',
+        'Jane',
+        'Doe'
+      );
```

Comment:
nit: It's probably preferable to refer to where the values come from than hardcoding the values again.  I think it helps with communicating intention and maintenance.